src/slowmokit/methods/metrics/classification_report.cpp

```diff
@@ -0,0 +1,149 @@
+/**
+ * @file methods/metrics/classification_report.hpp
+ *
+ * Easy include to show the classification report of data
+ */
+
+#include "classification_report.hpp"
+template <class T>
+
+ClassificationReport<T>::ClassificationReport(std::vector<T> &trueValue, std::vector<T> &predictedValue)
+{
+    this->trueValue = trueValue;
+    this->predictedValue = predictedValue;
+}
+
+template <class T>
+void ClassificationReport<T>::confusionMatrix(std::vector<T> &trueValue, std::vector<T> &predictedValue)
+{
+    int n = trueValue.size();
+
+    for (int i = 0; i < n; i++)
+    {
+        classes[trueValue[i]]++;
+        if (trueValue[i] == predictedValue[i])
+        {
+            truePositive[trueValue[i]]++;
+        }
+        if (trueValue[i] != predictedValue[i])
+        {
+            falseNegative[trueValue[i]]++;
+            falsePositive[predictedValue[i]]++;
+        }
+    }
+    // For True Negative, had to do it separately :crycatok:
+    for (auto it : classes)
+    {
+        int classNumber = it.first;
+        for (int i = 0; i < n; i++)
+        {
+            if (trueValue[i] != classNumber && predictedValue[i] != classNumber)
+            {
+                trueNegative[classNumber]++;
+            }
+        }
+    }
+}
+
+template <class T>
+std::map<T, double> ClassificationReport<T>::precision()
+{
+    // True Positive / (True Positive + False Positive)
+    std::map<T, double> precisionMap;
+    // std::cout << "class size"<<classes.size() << std::endl;
+    for (auto it : classes)
+    {
+        int classNumber = it.first;
+        // std::cout << classNumber << std::endl;
+        precisionMap[classNumber] = (double)(truePositive[classNumber] / (double)(truePositive[classNumber] + falsePositive[classNumber]));
```

Comment:
Ig this might give error if both TP and FP for the concerned class are 0. Similar case for recall too